### PR TITLE
Improve containers and NACHO 2.0.6

### PR DESCRIPTION
**Pull request by apeltzer**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/nanostring/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/nanostring _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by christopher-mohr** [comment]
> This is now ready for review @christopher-mohr

I will have a look later why the tests are failing.

**Comment by apeltzer** [comment]
Will just need this to pin matrixstats : https://github.com/BioContainers/multi-package-containers/pull/2880

**Comment by apeltzer** [comment]
@nf-core-bot fix linting pretty please

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -18,11 +18,11 @@
                 "python.linting.flake8Path": "/opt/conda/bin/flake8",
                 "python.linting.pycodestylePath": "/opt/conda/bin/pycodestyle",
                 "python.linting.pydocstylePath": "/opt/conda/bin/pydocstyle",
-                "python.linting.pylintPath": "/opt/conda/bin/pylint"
+                "python.linting.pylintPath": "/opt/conda/bin/pylint",
```
Comment (christopher-mohr) [inline .devcontainer/devcontainer.json:21]: was this added on purpose?

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Added the Bioinformatics publication to the citation list
 - Updated to nf-core template 2.11.1
+- Granularized the containers for NACHO and Heatmaps
```
Comment (christopher-mohr) [inline CHANGELOG.md:12]: We should probably also add this:

```suggestion
- Granularized the containers for NACHO and Heatmaps

### `Dependencies`
| Dependency | Old version | New version |
| ---------- | ----------- | ----------- |
| `nacho`    | 2.0.5       | 2.0.6       |
```

File: nextflow.config [hunk 3]
```diff
@@ -247,7 +247,7 @@ manifest {
     description     = """A Nanostring nCounter analysis pipeline"""
     mainScript      = 'main.nf'
     nextflowVersion = '!>=23.04.0'
-    version         = '1.2.1dev'
+    version         = '1.2.1'
```
Comment (christopher-mohr) [inline nextflow.config:250]: Still dev I guess?

```suggestion
    version         = '1.2.1dev'
```

File: assets/multiqc_config.yml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 report_comment: >
-  This report has been generated by the <a href="https://github.com/nf-core/nanostring/tree/dev" target="_blank">nf-core/nanostring</a> analysis pipeline. For information about how to interpret these results, please see the <a href="https://nf-co.re/nanostring/dev/docs/output" target="_blank">documentation</a>.
+  This report has been generated by the <a href="https://github.com/nf-core/nanostring/releases/tag/1.2.1" target="_blank">nf-core/nanostring</a> analysis pipeline. For information about how to interpret these results, please see the <a href="https://nf-co.re/nanostring/1.2.1/docs/output" target="_blank">documentation</a>.
```
Comment (christopher-mohr) [inline assets/multiqc_config.yml:2]: `dev` here as well?

**Review by christopher-mohr — CHANGES_REQUESTED** [review]
Just a couple of minor things.

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (apeltzer) [reply .devcontainer/devcontainer.json:21]: Newest linting changes require it yes

File: nextflow.config (hunk 3 above)
Comment (apeltzer) [reply nextflow.config:250]: No - will do immediate release after this

File: assets/multiqc_config.yml (hunk 4 above)
Comment (apeltzer) [reply assets/multiqc_config.yml:2]: Also no 😉

**Comment by christopher-mohr** [comment]
@nf-core-bot fix linting pretty please

**Review by christopher-mohr — APPROVED** [review]
LGTM

**Pull request merged**